### PR TITLE
Feature/dev 3220 add location to csv

### PR DESCRIPTION
**Pull request by tturnerswdev33**
## Briefly describe your changes
Added new column to MAP csv download that does a displayGeoName lookup to get full state or territory name
This code does not run when geoType = 'world' because Rob said that would break.

NOTE: I had some code I had added to chart for ticket 3221 that is related to Data Table header.  This code somehow got into this branch even though I had done a checkout of test and then git pull.  Dont know how.  So if you see an added useEffect in Chart you can just leave it.

## Checklist before requesting a review
- [x] My pull request was branched from and targets the test branch
- [x] I have performed a self-review of my code
- [x] I have manually tested all packages affected (bonus points for automations)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have checked to ensure there aren't other open pull requests for the same change

## Did you test your feature in the following environments?
- [x] Standalone Component
- [ ] Standalone Full Editor
- [ ] CDC Internal Checks

### CODE REVIEW

**Review by adamdoe — CHANGES_REQUESTED** [review]
### Please see the fixes and nits below
Note: I wasn't able to see the feature working. After fixing the below I'll take another pass on it. Lots of great work here, console.logs being removed, unused icons, etc. Mostly small feedback. Good work!

fix: Don't use hard-coded column names, ie `Fips Codes`
fix: Run prettier. If you can't fix the eslint issues, select the code and run the formatting manually.
fix: try to reduce some of the eslint errors in the file. I know you didn't create these, but we're trying to tackle some of them. This will help code formatting in the future for all devs.

nit: try to scope commits to tickets, not a big deal
nit: out of place comment

File: packages/chart/src/CdcChart.jsx [hunk 1]
```diff
@@ -486,6 +487,29 @@ export default function CdcChart({ configUrl, config: configObj, isEditor = fals
     }
   }, [config, stateData])
 
+  // DEV-3218 make "Chart Title" both a required field and default value
```
[the remark this answers is not in the document]
Comment (tturnerswdev33) [reply packages/chart/src/CdcChart.jsx:490]: Can you elaborate more specifically what you mean?

File: packages/map/src/components/DataTable.jsx [hunk 2]
```diff
@@ -150,8 +149,17 @@ const DataTable = props => {
   )
 
   const DownloadButton = memo(() => {
-    const csvData = Papa.unparse(rawData)
 
+    let csvData;
+    if (state.general.geoType === 'us-county' || state.general.geoType === 'single-state') {
+      // if county or single state have to look up based on fips codes
+      csvData = Papa.unparse(rawData.map(row => ({ FullGeoName: displayGeoName(row['FIPS Codes']), ...row })))
```
[the remark this answers is not in the document]
Comment (tturnerswdev33) [reply packages/map/src/components/DataTable.jsx:156]: A you referring to the made up column name "FullGeoName" or are you referring to using 'FIPS Codes'?
In either case not sure what you are asking me to do here... If the latter I dont see another way to get the data...without getting the specific col we need.

File: packages/map/src/components/DataTable.jsx [hunk 3]
```diff
@@ -150,8 +149,17 @@ const DataTable = props => {
   )
 
   const DownloadButton = memo(() => {
-    const csvData = Papa.unparse(rawData)
 
+    let csvData;
```
[the remark this answers is not in the document]
Comment (tturnerswdev33) [reply packages/map/src/components/DataTable.jsx:153]: Well I only had an if else... until testing showed there was another case that needed to be dealt with.

Yes you could do a switch but that would at a minimum double the lines required to do the exact same thing without much more clarity in my opinion.

**Comment by tturnerswdev33** [comment]
@adamdoe I dont see where the eslint issue is.  I have it say to format on save.  Which file are you referring to?

File: packages/map/src/components/DataTable.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamdoe) [reply packages/map/src/components/DataTable.jsx:156]: Column names are included in the config. Here you don't want to use `Fips Codes`. That's coming from the sample file you loaded. You'll want to browse the config to find where Fips Codes is being pulled from and use that value instead. For example, if you load a county data file and the columns are instead `County Name`, and `County FIps` the row you've hard coded here will never be found. Try looking for `state.columns.geo.name`.

**Comment by adamdoe** [comment]
@tturnerswdev33 - I ran through feedback here. In addition, if you need to run eslint you can always do that from the command line using `npx eslint [file_pattern]` but your editor should be able to provide the output in the problems tab when running the package.

**Comment by adamdoe** [comment]
@tturnerswdev33 - Also, when you're ready for this to be re-reviewed click the refresh button near the reviewers name. Thanks!

**Review by adamdoe — CHANGES_REQUESTED** [review]
I think I forgot to submit the review here. See feedback higher up on the chain.

File: packages/map/src/components/DataTable.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tturnerswdev33) [reply packages/map/src/components/DataTable.jsx]: Yep changed this

**Comment by adamdoe** [comment]
Breaking in the editor but not component.

**Review by adamdoe — CHANGES_REQUESTED** [review]
breaking in editor

**Pull request merged**